francis/noise_reduction.py

```diff
@@ -0,0 +1,44 @@
+import numpy as np
+import librosa
+import noisereduce
+
+
+def process(audio_data, seconds=2):
+    return noisereduce.reduce_noise(
+        audio_clip=audio_data[0],
+        noise_clip=__backround_noise(audio_data, seconds),
+        verbose=False,
+    )
+
+
+def __backround_noise(audio_data, seconds):
+    return __quietest(__split_audio(audio_data, seconds))
+
+
+def __quietest(split_amplitudes):
+    quietest_volume = None
+    quietest_index = 0
+    for i in range(len(split_amplitudes)):
+        average_volume = np.mean(np.abs(split_amplitudes[i]))
+        if quietest_volume == None or quietest_volume > average_volume:
+            quitest_volume = average_volume
+            quietest_index = i
+    return split_amplitudes[quietest_index]
+
```

Comment:
This takes the chunks of audio and finds the quietest chunk out of all of them, then the quietest chunk is returned